### PR TITLE
solve week 3

### PR DESCRIPTION
**Pull request by giaosudau**
Please review it for me!
Thanks.

### CODE REVIEW

File: week0003/giaosudau/python/week0003.py [hunk 1]
```diff
@@ -0,0 +1,67 @@
...
+
+    def __init__(self, name, amount):
+        self.name = name
+        self.amount = amount
+
+    def deposit(self, amount):
+        self.amount += amount
+        return True
+
+    def with_draw(self, amount):
```
Comment (vietlq) [inline week0003/giaosudau/python/week0003.py]: This is one word

File: week0003/giaosudau/python/week0003.py [hunk 2]
```diff
@@ -0,0 +1,67 @@
...
+        if amount <= self.amount:
+            self.amount -= amount
+            return True
+        else:
+            return False
+
+
+class Bank(object):
+    clients = {}
```
Comment (vietlq) [inline week0003/giaosudau/python/week0003.py]: Initialise it in the constructor. This belongs to class but not to instance

File: week0003/giaosudau/python/week0003.py [hunk 3]
```diff
@@ -0,0 +1,67 @@
+import sys
+
+
+class Client():
+    def __init__(self):
```
Comment (vietlq) [inline week0003/giaosudau/python/week0003.py]: Remove this

File: week0003/giaosudau/python/week0003.py [hunk 4]
```diff
@@ -0,0 +1,67 @@
...
+                print 'Invalid'
+
+    def deposit(self, name, amount):
+        if name in self.clients:
+            client = self.clients[name]
+            return client.deposit(amount)
+        else:
+            return False
+
+    def with_draw(self, name, amount):
```
Comment (vietlq) [inline week0003/giaosudau/python/week0003.py]: This is one word

File: week0003/giaosudau/python/week0003.py [hunk 5]
```diff
@@ -0,0 +1,67 @@
...
+            client = Client(name, amount)
+            self.clients[name] = client
+            return True
+
+    def process(self, input):
+        for line in input:
+            data = line.split(" ")
+            action = data[0]
+            name = data[1]
+            amount = int(data[2])
```
Comment (vietlq) [inline week0003/giaosudau/python/week0003.py]: The amont must be positive

File: week0003/giaosudau/python/week0003.py [hunk 6]
```diff
@@ -0,0 +1,67 @@
...
+            if action == 'A':
```
Comment (vietlq) [inline week0003/giaosudau/python/week0003.py]: Avoid magic strings. Use ACTION_ADD_CLIENT = 'A'

**Comment by giaosudau** [comment]
I fix it! 
Please review it one more time!
Thanks.

File: week0003/giaosudau/python/week0003.py [hunk 7]
```diff
@@ -0,0 +1,75 @@
...
+
+class Bank(object):
+    def __init__(self):
+        self.clients = {}
+
+    def process(self, input):
+        for line in input:
+            data = line.split(" ")
+            action = data[0]
+            name = data[1]
```
Comment (vietlq) [inline week0003/giaosudau/python/week0003.py]: The name must be handed in case-insensitive way

File: week0003/giaosudau/python/week0003.py [hunk 8]
```diff
@@ -0,0 +1,75 @@
...
+
+    def deposit(self, amount):
+        if amount > 0:
+            self.amount += amount
+            return True
+        else:
+            return False
+
+    def withdraw(self, amount):
+        if 0 < amount <= self.amount:
```
Comment (vietlq) [inline week0003/giaosudau/python/week0003.py]: Chained comparisons are good ;)

**Pull request merged**